### PR TITLE
CS-128 - Cardinal Quarter View and Decoupled Filter

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW

# Summary
 - Cardinal Quarter view with filters (PDB module and REACT for FE)
  - Ticket: https://stanfordits.atlassian.net/browse/CS-128
  - Project Charter: https://docs.google.com/document/d/1ah4k4xI4cNH8cEU_Zp2kXemQp9MS_c8WaztP0N5PBKg/edit?pli=1#heading=h.w59dx1ej63cp

# Review By (Date)
- ASAP! 

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Urgency
- How urgent is this? (Normal, High)

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Create a basic page and click on the "layout" tab.
4. I removed the breadcrumb, "Related Resources" and so on.
5. Create a new section and add the following block: "Cardinal Quarter Opportunities Filtering List"
6. Below that, create another section and add the view block called "Cardinal Quarter Opportunities: All Filtered"
7. Click save. 
Note: If useful, I have a page with the view blocks in place on the dev site: https://cardinalservice-dev.sites-pro.stanford.edu/node/31581/layout

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Comment by mdyoung3** [comment]
I'm going to branch off and work on the testing on another PR for now.

File: themes/cardinal_service/pdb_components/cardinal_quarter_opportunities_list/cardinal_quarter_opportunties_list.info.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Cardinal Quarter Opportunities Filtering List
+machine_name: cardinal-quarter-opportunities-filter-list
+type: pdb
+description: 'React ToDo'
+package: React
+core_version_requirement: ^9 || 10
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/cardinal_quarter_opportunities_list/cardinal_quarter_opportunties_list.info.yml]:
```suggestion
core_version_requirement: ^10 || ^11
```

File: themes/cardinal_service/pdb_components/opportunities_homepage/opportunties_homepage.info.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ machine_name: opportunities-homepage
 type: pdb
 description: 'React filters for the CS homepage'
 package: React
-core_version_requirement: ^8.8 || ^9
+core_version_requirement: ^9 || ^10
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/opportunities_homepage/opportunties_homepage.info.yml]:
```suggestion
core_version_requirement: ^10 || ^11
```

File: themes/cardinal_service/pdb_components/opportunities_list/opportunties_list.info.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ machine_name: opportunities-filter-list
 type: pdb
 description: 'React ToDo'
 package: React
-core_version_requirement: ^8.8 || ^9
+core_version_requirement: ^9 || ^10
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/opportunities_list/opportunties_list.info.yml]:
```suggestion
core_version_requirement: ^10 || ^11
```

File: themes/cardinal_service/pdb_components/resources/resources.info.yml [hunk 4]
```diff
@@ -3,7 +3,7 @@ machine_name: resources-filters
 type: pdb
 description: 'React filters for the Resources content'
 package: React
-core_version_requirement: ^8.8 || ^9
+core_version_requirement: ^9 || ^10
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/resources/resources.info.yml]:
```suggestion
core_version_requirement: ^10 || ^11
```

File: config/sync/system.logging.yml [hunk 5]
```diff
@@ -1,3 +1,3 @@
 _core:
   default_config_hash: u3-njszl92FaxjrCMiq0yDcjAfcdx72w1zT1O9dx6aA
-error_level: hide
+error_level: all
```
Comment (pookmish) [inline config/sync/system.logging.yml]: No need to change this file.

File: config/sync/config_ignore.settings.yml [hunk 6]
```diff
@@ -2,24 +2,24 @@ _core:
...
-  - 'core.extension:theme'
-  - 'system.theme:default'
-  - '~block.block.claro_*'
-  - '~block.block.stanford_profile_admin_*'
-  - '~block.block.stanford_basic_*'
-  - '~block.block.cardinal_service_*'
-  - '~block.block.minimally_branded_subtheme_*'
   - 'block.block.*'
   - 'core.extension:theme'
+  - 'core.extension:theme'
```
Comment (pookmish) [inline config/sync/config_ignore.settings.yml]: Looks like a duplicate line we can remove.

**Review by pookmish — CHANGES_REQUESTED** [review]
It's unclear to me what I should expect to see. Instructions are not very clear and don't provide any actual steps to verify. Please provide more helpful information or links to designs or something. Even the Jira ticket has no information. I have no context.

File: config/sync/config_ignore.settings.yml [hunk 7]
```diff
@@ -2,13 +2,6 @@ _core:
   default_config_hash: v7E8C8wJWeAW2BGohMNY1tZSBc4bexM6O62tGxecTfE
 mode: simple
 ignored_config_entities:
-  - 'core.extension:theme'
-  - 'system.theme:default'
-  - '~block.block.claro_*'
-  - '~block.block.stanford_profile_admin_*'
-  - '~block.block.stanford_basic_*'
-  - '~block.block.cardinal_service_*'
-  - '~block.block.minimally_branded_subtheme_*'
```
Comment (pookmish) [inline config/sync/config_ignore.settings.yml]: We shouldn't be removing these

File: themes/cardinal_service/pdb_components/Components/Filters.jsx [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import React, {Component} from 'react';
+import React, {Component, useEffect} from 'react';
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/Components/Filters.jsx]: `useEffect` isn't used

File: themes/cardinal_service/pdb_components/Components/Filters.jsx [hunk 9]
```diff
@@ -141,14 +143,15 @@ export class Filters extends Component {
...
-
   }
 
   /**
    * Select list event action passed down to the select component.
    */
   onSelectChange(fieldName, selectedValues) {
     const newState = {...this.state};
+    console.log('fieldname: ' + fieldName);
+    console.log('Selected value: ' + selectedValues);
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/Components/Filters.jsx]: remove console logs

**Review by pookmish — CHANGES_REQUESTED** [review]

File: config/sync/config_ignore.settings.yml (hunk 7 above)
Comment (mdyoung3) [reply config/sync/config_ignore.settings.yml]: @pookmish I confirmed on my local these are in there. In fact, it's in there twice. Not sure what's going on with that. 

Maybe I should just pull the 11.x version and push it. Pretty sure I didn't edited this.

File: themes/cardinal_service/pdb_components/Components/Filters.jsx (hunk 8 above)
Comment (mdyoung3) [reply themes/cardinal_service/pdb_components/Components/Filters.jsx]: Removed

File: themes/cardinal_service/pdb_components/Components/Filters.jsx (hunk 9 above)
Comment (mdyoung3) [reply themes/cardinal_service/pdb_components/Components/Filters.jsx]: Gone

File: themes/cardinal_service/pdb_components/Components/Filters.jsx [hunk 10]
```diff
@@ -113,10 +113,12 @@ export class Filters extends Component {
    */
   componentDidMount() {
     const that = this;
+    const params = this.props.apiParams ? "?" + queryString.stringify(this.props.apiParams, {arrayFormat: 'bracket'}) : ""
 
-    fetch('/api/terms-used/' + this.props.bundle)
+    fetch('/api/terms-used/' + this.props.bundle + params)
       .then((response) => response.json())
       .then((jsonData) => {
+        console.log(jsonData);
```
Comment (pookmish) [inline themes/cardinal_service/pdb_components/Components/Filters.jsx]: oh, one more console log

**Review by pookmish — APPROVED** [review]
Merge at your convenience.

**Comment by mdyoung3** [comment]
Thanks, @pookmish !

**Pull request merged**